core/src/main/java/org/pentaho/di/core/plugins/PluginRegistry.java

```diff
@@ -463,7 +463,14 @@ private String createSupplemantalKey( String pluginName, String id ) {
     }
 
     if ( plugin instanceof ClassLoadingPluginInterface ) {
-      return ( (ClassLoadingPluginInterface) plugin ).loadClass( pluginClass );
+      T aClass = ( (ClassLoadingPluginInterface) plugin ).loadClass( pluginClass );
+      if ( aClass == null ) {
+        throw new KettlePluginClassMapException( BaseMessages
+            .getString( PKG, "PluginRegistry.RuntimeError.NoValidClassRequested.PLUGINREGISTRY002", plugin.getName(),
+                pluginClass.getName() ) );
+      } else {
```

Comment:
else is not needed here.

Reply:
Can't agree. Else branch is how it worked before (return plugin.loadClass), without it we're not getting services plugin provides.

Reply:
Sorry didn't see the code below. Thanks 👍